### PR TITLE
8255606: Enable concurrent stack processing on x86_32 platforms

### PR DESCRIPTION
**Pull request by zhengyu123**
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Testing

|     | Linux x64 | Linux x86 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- | ----- |
| Build | ✔️ (5/5 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) |
| Test (tier1) | ✔️ (9/9 passed) | ✔️ (9/9 passed) | ✔️ (9/9 passed) | ✔️ (9/9 passed) |

### Issue
 * [JDK-8255606](https://bugs.openjdk.java.net/browse/JDK-8255606): Enable concurrent stack processing on x86_32 platforms

### Reviewers
 * [Aleksey Shipilev](https://openjdk.java.net/census#shade) (@shipilev - **Reviewer**)
 * [Roman Kennke](https://openjdk.java.net/census#rkennke) (@rkennke - **Reviewer**)
 * [Erik Österlund](https://openjdk.java.net/census#eosterlund) (@fisk - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/945/head:pull/945`
`$ git checkout pull/945`

### CODE REVIEW

File: src/hotspot/cpu/x86/x86_32.ad [hunk 1]
```diff
@@ -653,8 +653,9 @@ void MachEpilogNode::format( PhaseRegAlloc *ra_, outputStream* st ) const {
   }
   st->print_cr("POPL   EBP"); st->print("\t");
   if (do_polling() && C->is_method_compilation()) {
-    st->print("TEST   PollPage,EAX\t! Poll Safepoint");
-    st->cr(); st->print("\t");
+    st->print("cmpptr   rsp, poll_offset[thread]  \n\t"
```
Comment (shipilev) [inline src/hotspot/cpu/x86/x86_32.ad:656]: Let's say `CMPL` here? I don't think we reference `cmpptr` in outputs here. Also, note every instruction is capitalized in `x86_32.ad`.

File: src/hotspot/cpu/x86/x86_64.ad [hunk 2]
```diff
@@ -930,8 +930,8 @@ void MachEpilogNode::format(PhaseRegAlloc* ra_, outputStream* st) const
   st->print_cr("popq    rbp");
   if (do_polling() && C->is_method_compilation()) {
     st->print("\t");
-    st->print_cr("cmpq    poll_offset[r15_thread], rsp\n\t"
-                 "ja      #safepoint_stub\t"
+    st->print_cr("cmpptr   rsp, poll_offset[r15_thread] \n\t"
```
Comment (shipilev) [inline src/hotspot/cpu/x86/x86_64.ad:933]: Ditto, leave `cmpq` here.

**Review by shipilev — CHANGES_REQUESTED** [review]
I have minor comments.

File: src/hotspot/cpu/x86/x86_32.ad (hunk 1 above)
Comment (zhengyu123) [reply src/hotspot/cpu/x86/x86_32.ad]: Thanks for the review.  Updated according to your suggestions.

**Review by rkennke — APPROVED** [review]
Looks good to me! Thanks!

**Review by fisk — APPROVED** [review]
Looks good. Nice to see one more platform support this. Hoping that eventually all platforms will support this.

**Comment by zhengyu123** [comment]
> Looks good. Nice to see one more platform support this. Hoping that eventually all platforms will support this.

Thanks, Erik.

**Review by shipilev — APPROVED** [review]

**Pull request closed without merging**